src/hunter/__init__.py

```diff
@@ -271,6 +272,32 @@ def From(condition=None, predicate=None, watermark=0, **kwargs):
         return _From(condition, predicate, watermark)
 
 
+def Backlog(condition=None, action=CallPrinter, size=100, stack_depth=0, **kwargs):
+    condition = _get_condition(condition, kwargs)
+    if inspect.isclass(action):
+        action = action()
+
+    return _Backlog(condition, action, size, stack_depth)
+
+
+def _backlog_filter(self, condition=None, **kwargs):
+    self.filter_condition = _get_condition(condition, kwargs)
+    return self
+
+
+_Backlog.filter = _backlog_filter
```

Comment:
We can't use tricks like this as it would prevent a certain class of optimizations in the future Cython implementation. Everything needs to be implemented in the predicate class (a filter method that just copies itself +  the filter args - it only needs to import Q locally).